File: BojS1_11403.py
# BojS1_11403_경로 찾기
import sys

global n, g, visit


def solution():
    global n, g, visit
    n = int(sys.stdin.readline())
    g = [list(map(int, sys.stdin.readline().split())) for _ in range(n)]
    visit = [0 for _ in range(n)]

    for i in range(n):
        dfs(i)
        print(*visit)
        visit = [0 for _ in range(n)]


def dfs(x):
    global visit
    for j in range(n):
        if visit[j] == 0 and g[x][j] == 1:
            visit[j] = 1
            dfs(j)


if __name__ == '__main__':
    solution()

File: BojS1_11403.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.StringTokenizer;

public class BojS1_11403 {    // BojS1_11403_경로 찾기

    int n;
    ArrayList<ArrayList<Integer>> g;
    ArrayList<Integer> visit;

    private void solution() throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        n = Integer.parseInt(br.readLine());
        g = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            ArrayList<Integer> subList = new ArrayList<>();
            for (int j = 0; j < n; j++) {
                subList.add(Integer.parseInt(st.nextToken()));
            }
            g.add(subList);
        }

        initVisitZero();
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < n; i++) {
            dfs(i);

            for (int value : visit) {
                sb.append(value).append(' ');
            }
            sb.append('\n');

            initVisitZero();
        }
        System.out.print(sb);
    }

    private void dfs(int x) {
        for (int i = 0; i < n; i++) {
            if (g.get(x).get(i) == 1 && visit.get(i) == 0) {
                visit.set(i, 1);
                dfs(i);
            }
        }
    }

    private void initVisitZero() {
        visit = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            visit.add(0);
        }
    }

    public static void main(String[] args) throws IOException {
        new BojS1_11403().solution();
    }

}
